components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1756,10 +1756,18 @@ public int compare(Tag o1, Tag o2) {
 
             Map<String, Tag> tagsData = new HashMap<String, Tag>();
             try {
-            	PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(((UserRegistry)userRegistry).getUserName());
+                String username = ((UserRegistry)userRegistry).getUserName();
+                String tenantDomain = APIUtil.getTenantDomainFromTenantId(tenantId);
+                //add the tenant domain to the username if it is not present in the case of email as the username
+                username = APIUtil.appendTenantDomainForEmailUsernames(username, tenantDomain);
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(username);
```

Comment:
If `requestedTenant != null`, this code is useless right? because it's being overwritten by the below code.

Reply:
This change is removed as it is not needed.